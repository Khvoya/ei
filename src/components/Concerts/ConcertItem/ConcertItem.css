.ConcertItem {
  display: flex;
  cursor: pointer;
  flex-direction: row;
  padding: 20px 30px;
  justify-content: space-between;
  background-color: white;
  border: solid black 2px;
}

.ConcertItem div{
  width: 100%;
}

.ConcertItem__link,
.ConcertItem__link:hover,
.ConcertItem__link:active {
  color: black;
}

.ConcertItem:nth-last-child(1) {
  border-bottom-left-radius: 20px;
  border-bottom-right-radius: 20px;
}
.ConcertItem:nth-child(1) {
  border-top-left-radius: 20px;
  border-top-right-radius: 20px;
}

@media screen and (max-width: 420px) {
  .ConcertItem {
    padding: 10px 20px;
  }
}
