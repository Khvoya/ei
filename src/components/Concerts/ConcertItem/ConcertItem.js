import React, { Component } from "react";
import "./ConcertItem.css";

class ConcertItem extends Component {
    render() {
        const props = this.props;
        return (
            <div className="ConcertItem">
                <div className="ConcertItem__date">{props.date}</div>
                <div className="ConcertItem__country">{props.country}</div>
                <div className="ConcertItem__club">{props.club}</div>
            </div>
        );
    }
}

export default ConcertItem;
