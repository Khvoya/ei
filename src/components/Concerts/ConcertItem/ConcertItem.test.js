import React from "react";
import { shallow } from "enzyme";
import ConcertItem from "./ConcertItem";

describe("ConcertItem", () => {
  const props = {
    club: "Port Sevkabel",
    country: "Russia, Saint-Petersburg",
    date: "09.06.2019",
    link: "https://vk.com/duhulits2"
  };

  it("should render correctly", () => {
    const output = shallow(<ConcertItem  {...props}/>);
    expect(output).toMatchSnapshot();
  });
});
