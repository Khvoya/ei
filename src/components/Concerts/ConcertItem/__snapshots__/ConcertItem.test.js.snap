// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConcertItem should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConcertItem
    club="Port Sevkabel"
    country="Russia, Saint-Petersburg"
    date="09.06.2019"
    link="https://vk.com/duhulits2"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          09.06.2019
        </div>,
        <div
          className="ConcertItem__country"
        >
          Russia, Saint-Petersburg
        </div>,
        <div>
          Port Sevkabel
        </div>,
      ],
      "className": "ConcertItem__link ConcertItem",
      "href": "https://vk.com/duhulits2",
      "target": "blank",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "09.06.2019",
        },
        "ref": null,
        "rendered": "09.06.2019",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Russia, Saint-Petersburg",
          "className": "ConcertItem__country",
        },
        "ref": null,
        "rendered": "Russia, Saint-Petersburg",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Port Sevkabel",
        },
        "ref": null,
        "rendered": "Port Sevkabel",
        "type": "div",
      },
    ],
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            09.06.2019
          </div>,
          <div
            className="ConcertItem__country"
          >
            Russia, Saint-Petersburg
          </div>,
          <div>
            Port Sevkabel
          </div>,
        ],
        "className": "ConcertItem__link ConcertItem",
        "href": "https://vk.com/duhulits2",
        "target": "blank",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "09.06.2019",
          },
          "ref": null,
          "rendered": "09.06.2019",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Russia, Saint-Petersburg",
            "className": "ConcertItem__country",
          },
          "ref": null,
          "rendered": "Russia, Saint-Petersburg",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Port Sevkabel",
          },
          "ref": null,
          "rendered": "Port Sevkabel",
          "type": "div",
        },
      ],
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
